doc/whats_new/v0.21.rst

```diff
@@ -429,6 +429,12 @@ Support for Python 3.4 and below has been officially dropped.
   excluded in a cross-validation fold.
   :issue:`13366` by :user:`Guillaume Fournier <gfournier>`
 
+- |Enhancement| Both :class:`~model_selection.GridSearchCV` and
+  :func:`~model_selection.cross_val_score`
+  (:func:`~model_selection.cross_validate` as well) now accept a sparse ``y``
+  (with shape ``(n_samples, 1)`` or ``(n_samples,)``). :issue:`4225` by
```

Comment:
If we were to accept a sparse `y` with shape `(n_samples, n_labels)` and `n_labels > 1` (`type_of_target(y)` is `multilabel-indicator`), this wouldn't work with `StratifiedKFold` or `StratifiedShuffleSplit` since it expects `y` to be a column vector... This would work by converting `y` from `multilabel_indicator` to `multiclass`. [this may also answer your comment below]